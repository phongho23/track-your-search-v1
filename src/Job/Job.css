.Job {
    --Job-height: 88px;
    background-color: #505256;
    color: white;
    height: var(--Job-height);
    margin-bottom: 8px;
    max-width: 380px;
    padding: 8px 0 8px 8px;
    position: relative;
  }
  
  .Job__delete {
    --Job-button-height: 32px;
    background-color: #041a3a;
    border: 0;
    color: red;
    float: right;
    font-size: 12px;
    height: var(--Job-button-height);
    position: relative;
    top: calc((var(--Job-button-height) + 32px) * -1);
    transform-origin: right bottom;
    transform: rotate(-90deg);
    width: calc(var(--Job-height));
  }
  .Job__delete:hover {
    background-color: #505256;
    color: red;
    cursor: pointer;
    font-weight: bold;
  }
  
  .Job__title {
    font-family: var(--custom-font);
    font-size: 20px;
    margin: 0;
    max-width: 80%;
    min-height: 24px;
  }
  .Job__title a {
    color: white;
    display: block;
    text-decoration: none;
  }
  .Job__title a:hover {
    color: #DB7676;
  }
  
  .Job__dates {
    bottom: 8px;
    color: white;
    font-size: 14px;
    position: absolute;
  }