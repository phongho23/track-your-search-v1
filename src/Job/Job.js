
import React from 'react'
import { Link, Redirect } from 'react-router-dom'
import { format } from 'date-fns'
import './Job.css'
import ApiContext from '../ApiContext'
import config from '../config'

export default class Job extends React.Component {
  static defaultProps = {
    onDeleteJob: () => {},
  }
  static contextType = ApiContext;

  handleClickDelete = e => {
    e.preventDefault()
    const jobId = this.props.id

    fetch(`${config.API_ENDPOINT}/jobs/${jobId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
      .then(() => {
        this.context.deleteJob(jobId)
        this.props.onDeleteJob(jobId)
      })
      .catch(error => {
        console.error({ error })
      })
  }

render() {
  const { jobtitle, id, modified } = this.props

  return (
    <div style={
      {width: "150px", 
        height: "150px", 
        margin: "10px", 
        position: "relative"}} 
        className='Job'
    >
      <h2 className='Job__title'>
        <Link to={`/home/job/${id}`}>
          {jobtitle}
        </Link>
      </h2>
      <button style={{color: "red", fontSize: "20px", position: "absolute", top: "0", right: "0" }} 
      type='button'
      onClick={this.handleClickDelete}
      >
        {' '}
        X
      </button>
      <div className='Job__dates'>
        <div className='Job__dates-modified'>
          Modified
          {' '}
          <span className='Date'>
            {format(modified, 'Do MMM YYYY')}
          </span>
        </div>
      </div>
    </div>
    )
  }
}