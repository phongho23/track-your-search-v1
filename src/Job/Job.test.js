import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Job from './Job'

describe(`Job component`, () => {
  const props = {
    id: '1',
    name: 'test-class-name',
    modified: '2021-01-03T00:00:00.000Z',
  }

  it('renders a .Job by default', () => {
    const wrapper = shallow(<Job />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Job given props', () => {
    const wrapper = shallow(<Job {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
