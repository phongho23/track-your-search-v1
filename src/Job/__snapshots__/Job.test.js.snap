// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job component renders a .Job by default 1`] = `
<div
  className="Job"
  style={
    Object {
      "height": "150px",
      "margin": "10px",
      "position": "relative",
      "width": "150px",
    }
  }
>
  <h2
    className="Job__title"
  >
    <Link
      to="/home/job/undefined"
    />
  </h2>
  <button
    onClick={[Function]}
    style={
      Object {
        "color": "red",
        "fontSize": "20px",
        "position": "absolute",
        "right": "0",
        "top": "0",
      }
    }
    type="button"
  >
     
    X
  </button>
  <div
    className="Job__dates"
  >
    <div
      className="Job__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        Invalid Date
      </span>
    </div>
  </div>
</div>
`;

exports[`Job component renders the Job given props 1`] = `
<div
  className="Job"
  style={
    Object {
      "height": "150px",
      "margin": "10px",
      "position": "relative",
      "width": "150px",
    }
  }
>
  <h2
    className="Job__title"
  >
    <Link
      to="/home/job/1"
    />
  </h2>
  <button
    onClick={[Function]}
    style={
      Object {
        "color": "red",
        "fontSize": "20px",
        "position": "absolute",
        "right": "0",
        "top": "0",
      }
    }
    type="button"
  >
     
    X
  </button>
  <div
    className="Job__dates"
  >
    <div
      className="Job__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        2nd Jan 2021
      </span>
    </div>
  </div>
</div>
`;
